debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlValueConverters.java

```diff
@@ -528,23 +545,52 @@ protected Object convertPoint(Column column, Field fieldDefn, Object data){
 
         if (data == null) {
             if (column.isOptional()) return null;
-            return io.debezium.data.geometry.Point.createValue(schema, 0.0, 0.0);
+            throw new IllegalArgumentException("Nulls not valid on " + column);
```

Comment:
A point at (0,0) is _not the same_ as "no point"... see [Null Island](https://en.wikipedia.org/wiki/Null_Island) 😄 

One option is an "empty" geometry — conceptually "a valid geometry but without any data" (example: "the intersecting area of two polygons that don't intersect" — `A ∩ B` would be NULL if A and/or B are NULL, but would be an "empty polygon" if A doesn't intersect B). In the OGC Simple Features for SQL model there is `POLYGON EMPTY`, `GEOMETRYCOLLECTION EMPTY`, etc.

But there are a bunch of compatibility issues & edge cases with empty geometries across different software IME. But can implement if you think that's better :)

I assume the "default values for everything" is trying to fit a model where the constraint is not-null but somehow we received a record without data for a required field? Shouldn't that raise an error rather than making assumptions everywhere? I mean, it must for any other DB constraint more complex than "not null"?

Reply:
Added the generic `GEOMETRYCOLLECTION EMPTY` for Postgis+MySQL. 

_Doesn't_ apply to the legacy Point objects, because they have X&Y fields as well, which can't be null.